### ISSUE TITLE
Crash when launching Tiled 1.10.2 installed from the itch.io app

### ISSUE THREAD

**Issue by carbonacat** [mapeditor/tiled#3831]
**Describe the bug**

**To Reproduce**
- Install and configure itch.io app for mac OS
- Install Tiled for macOS 11+ 1.10.2
- Attempt to launch it (either from the itch.io app or directly by opening the Tiled.app)
- A standard crash report "Tiled.app quit unexpectedly." will appear 100% of time, with such a report :

[20231017-tiled1.10.2-crashreport.txt](https://github.com/mapeditor/tiled/files/12926210/20231017-tiled1.10.2-crashreport.txt)

**Expected behavior**
- Tiled launches properly.

**Media**
![image](https://github.com/mapeditor/tiled/assets/22575458/d50338a7-9f15-4a91-babd-ecc7ee70c248)

**Specifications:**
 - OS: macOS Monterey 12.6.7
 - Tiled Version: 1.10.2
 - itch.io Version: 25.6.2

**Additional Notes:**
- It doesn't seem to happen with the released available here on github.
- Reverting to itch.io's Tiled 1.10.1 version works properly. Only this 1.10.2 version behaves this way so far I tested.
- Manually deleting Tiled.app from itch.io app's location then repairing/reinstalling it will make it work properly.

**Comment by bjorn** [comment]
Hmm, the reason for this crash appears to be a missing library, or a missing or wrong code signature:

```
Termination Reason:    Namespace DYLD, Code 1 Library missing
Library not loaded: '@rpath/QtGui.framework/Versions/A/QtGui'
Referenced from: '/Users/USER/Library/Application Support/itch/*/Tiled.app/Contents/MacOS/Tiled'
Reason: tried: '' (code signature does not cover entire file up to signature), '' (code signature does not cover entire file up to signature), '' (no such file), '' (no such file)
(terminated at launch; ignore backtrace)
```

> Manually deleting Tiled.app from itch.io app's location then repairing/reinstalling it will make it work properly.

Does that apply even with the 1.10.2 version? Because that implies a bug in the itch.io app rather than in the Tiled release...

**Comment by carbonacat** [comment]
> Does that apply even with the 1.10.2 version? Because that implies a bug in the itch.io app rather than in the Tiled release...

I think you meant 1.10.1 so I tested switching back to 1.10.1 from an installed 1.10.2 and it crashes the same exact way.
It does look like it's related to itch.io's app.

I'll open an issue here as well - I'm not sure what to do about this issue here. People on Monterey will probably encounter this issue after updating Tiled to the universal binary.

**Comment by bjorn** [comment]
> I'll open an issue here as well - I'm not sure what to do about this issue here. People on Monterey will probably encounter this issue after updating Tiled to the universal binary.

It's likely, though not many seem to be using the itch.io app then. The 1.10.2 release is a while ago already and I didn't get this report before (not in this level of detail, at least). But thanks for opening that issue!

**Comment by carbonacat** [comment]
> Judging by the development activity on the itch.io app though, it does not look like we can expect much action...

I agree, especially since it's kind of a niche bug which is easily circumvented by uninstalling/installing cycle or manual deletion. Well at least it's documented, thanks for your reactivity!
